### ISSUE TITLE
Strange version resolving

### ISSUE THREAD

**Issue by sc0rp10** [yarnpkg/yarn#1158]
**Do you want to request a *feature* or report a *bug*?**
Bug
**What is the current behavior?**
package.json
```json
{
  "dependencies": {
    "devbridge-autocomplete": "^1.2.26",
    "jquery": "2.1.2"
  }
}
```
Will install:
```yaml
devbridge-autocomplete:
  version "1.2.26"
  resolved "https://registry.yarnpkg.com/devbridge-autocomplete/-/devbridge-autocomplete-1.2.26.tgz#e491e01b701ba0ff509afaf4ee1eca69a9d00647"
  dependencies:
    jquery ">=1.7"

jquery@>=1.7:
  version "3.1.1"
  resolved "https://registry.yarnpkg.com/jquery/-/jquery-3.1.1.tgz#347c1c21c7e004115e0a4da32cece041fad3c8a3"

jquery@2.1.2:
  version "2.1.2"
  resolved "https://registry.yarnpkg.com/jquery/-/jquery-2.1.2.tgz#b68f154cb2ea4731924883e9fe20ec199d1dc1e2"


```
**What is the expected behavior?**
Install jquery once with version = 2.1.2
**Please mention your node.js, yarn and operating system version.**
```
$ node -v
v6.7.0
$ yarn -v
yarn install v0.15.1
```

**Comment by davidmerrique** [comment]
Same problem here

```json
{
  "dependencies": {
    "jquery": "2.2.4",
    "parsleyjs": "2.5.1"
  }
}
```

```yaml
jquery@>=1.8.0:
  version "3.1.1"
  resolved "https://registry.yarnpkg.com/jquery/-/jquery-3.1.1.tgz#347c1c21c7e004115e0a4da32cece041fad3c8a3"

jquery@2.2.4:
  version "2.2.4"
  resolved "https://registry.yarnpkg.com/jquery/-/jquery-2.2.4.tgz#2c89d6889b5eac522a7eea32c14521559c6cbf02"

parsleyjs@2.5.1:
  version "2.5.1"
  resolved "https://registry.yarnpkg.com/parsleyjs/-/parsleyjs-2.5.1.tgz#6380924e231c38b6eca3a597d7ebc0dde41ceb9f"
  dependencies:
    jquery ">=1.8.0"
```

Parsley depends on `jquery@>=1.8.0`, but `3.1.1` is still being installed even though `2.2.4` should satisfy it.

**Comment by vladferix** [comment]
+1

**Comment by fearmear** [comment]
npm example
```
> mkdir npm-test && cd npm-test
> npm init --yes
> npm install moment@2.9.0 angular-moment@1.0.0
npm-test@1.0.0 /npm-test
├── angular-moment@1.0.0
└── moment@2.9.0
```

yarn example
```
> mkdir yarn-test && cd yarn-test
> yarn add moment@2.9.0 angular-moment@1.0.0
yarn add v0.16.0
info No lockfile found.
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 📃  Building fresh packages...
success Saved lockfile.
success Saved 3 new dependencies.
├─ angular-moment@1.0.0
│  └─ moment@2.15.1
└─ moment@2.9.0
✨  Done in 6.26s.
```

**Comment by kittens** [comment]
Fixed via #1477.
